### PR TITLE
resourcemanager: do not check existence when add resource group

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: ref #5851, ref pingcap/tidb#45050

### What is changed and how does it work?

```
Do not check the existence of target resource group to make create resource group idempotent.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/integrations/mcs/resourcemanager/resource_manager_test.go [hunk 1]
```diff
@@ -789,7 +789,7 @@ func (suite *resourceManagerClientTestSuite) TestBasicResourceGroupCURD() {
 		}
 		// Create Resource Group
 		resp, err := cli.AddResourceGroup(suite.ctx, group)
-		checkErr(err, tcase.addSuccess)
+		checkErr(err, true)
 		if tcase.addSuccess {
```
Comment (HuSharp) [inline tests/integrations/mcs/resourcemanager/resource_manager_test.go:793]: Maybe it's better to change `addSuccess` name to meet now the test AddResourceGroup behavior will actually succeed.

Comment (glorv) [reply tests/integrations/mcs/resourcemanager/resource_manager_test.go]: done

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by CabinfeverB — APPROVED** [review]
Can we add some comment for this behavior? REST LGTM

File: pkg/mcs/resourcemanager/server/manager.go [hunk 2]
```diff
@@ -163,12 +163,6 @@ func (m *Manager) AddResourceGroup(grouppb *rmpb.ResourceGroup) error {
 	if grouppb.GetPriority() > 16 {
 		return errs.ErrInvalidGroup
 	}
-	m.RLock()
-	_, ok := m.groups[grouppb.Name]
-	m.RUnlock()
-	if ok {
-		return errs.ErrResourceGroupAlreadyExists.FastGenByArgs(grouppb.Name)
```
Comment (HuSharp) [inline pkg/mcs/resourcemanager/server/manager.go:170]: Need to remove the `ErrResourceGroupAlreadyExists` related error definition.

Comment (glorv) [reply pkg/mcs/resourcemanager/server/manager.go:170]: done.

**Comment by lhy1024** [comment]
Do we need pick it to 7.5 and 8.1?

**Comment by glorv** [comment]
> Do we need pick it to 7.5 and 8.1?

@lhy1024 This PR is merge into merge before the release-7.5 branch is created

**Comment by lhy1024** [comment]
> > Do we need pick it to 7.5 and 8.1?
> 
> @lhy1024 This PR was merge into merge before the release-7.5 branch is created. Did you refer to the wrong PR?

Because I see affect-7.5 and affect-8.1 on https://github.com/pingcap/tidb/issues/45050. Could you remove these labels on this issue?

**Pull request merged**